bridge/elrond/client.go

```diff
@@ -110,23 +110,24 @@ func (c *client) poll(ctx context.Context) {
 	for {
 		select {
 		case <-time.After(c.nonceUpdateInterval):
-			c.saveCurrentNonce()
+			c.updateCurrentNonce()
 		case <-ctx.Done():
 			c.log.Debug("Client.poll function is closing...")
 			return
 		}
 	}
 }
 
-func (c *client) saveCurrentNonce() {
+func (c *client) updateCurrentNonce() {
 	account, err := c.proxy.GetAccount(c.address)
 	if err != nil {
 		c.log.Debug("Elrond: error polling account", "address", c.address.AddressAsBech32String(), "error", err.Error())
 		return
 	}
 
-	c.log.Debug("Elrond: polled account", "address", c.address.AddressAsBech32String(), "nonce", account.Nonce)
-	atomic.StoreUint64(&c.nonce, account.Nonce)
+	if c.nonce < account.Nonce {
```

Comment:
@iulianpascalau as discussed this does not take into account when blocks into a shard get reverted, in this case the account nonce will indeed be lower,

Not sure how to resolve this at this time, a very simple solution will be if the gap between the 2 nonce is 'unusually big' then store the account nonce.

Reply:
There is one problem here, if, for example, the releayer's stored nonces are higher with one value above the account's nonce. The relayers will produce transactions that will never be executed and automatically deleted after 100 rounds (~10 minutes).

Reply:
What if instead of just a nonce, we have an association between a nonce and a counter?
Then, each time we have `c.nonce == account.nonce` we increase the counter.
When we have `c.nonce != account.nonce` we store the new nonce and reset counter.

Each sensible action will happen if `counter >= k`

Reply:
Had a productive meeting with @constantintovisi and will start ASAP the creation of a new component in erd-go sdk that will handle the nonce increment. Spoiled: it will have also a retrial mechanism in-place.

Reply:
Cool, then my guess is that https://github.com/ElrondNetwork/elrond-eth-bridge/issues/57 will also be obsolete.